### PR TITLE
#44 Result

### PR DESCRIPTION
**Pull request by VsSekorin**
As per #44 
Result introduced

### CODE REVIEW

**Comment by 0crat** [comment]
@vssekorin I'm not managing this repo, remove the [webhook](https://github.com/yegor256/cactoos/settings/hooks) or contact me in [Slack](http://www.zerocracy.com) //cc @yegor256

File: src/main/java/org/cactoos/func/Result.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+     */
+    private final Func<X, Y> func;
+    /**
+     * The argument.
+     */
+    private final X arg;
+    /**
+     * The result.
+     */
+    private final List<Y> value;
```
Comment (dusan-rychnovsky) [inline src/main/java/org/cactoos/func/Result.java]: Why a list if you only ever need the first element?

Comment (VsSekorin) [inline src/main/java/org/cactoos/func/Result.java]: This is caching. Most likely it must be carried out in the decorator. But it seems to me that the object does not make sense without it.
How to implement caching differently with immutable objects?

**Comment by yegor256** [comment]
@VsSekorin I don't think I like the design here. We already have `StickyFunc`. We have to reuse it. How about this:

```java
new FuncAsValue(
  new StickyFunc(func),
  arg
);
```

This `FuncAsValue` will `implement Func.Value`.

**Pull request closed without merging**